src/test/java/com/materialidentity/schemaservice/RenderTest.java

```diff
@@ -206,24 +225,154 @@ void CoA_renderEndpointTest_InvalidSchemaType() throws Exception {
 				.jsonPath("$.message").isEqualTo("Certificate type 'invalid' is not supported.");
 	}
 
-	private void assertPdfContentEquals(byte[] expectedPdfContent, byte[] actualPdfContent) throws Exception {
+	private void assertPdfContentEquals(byte[] expectedPdfContent, byte[] actualPdfContent, 
+			String fileName) throws Exception {
 		PdfReader expectedReader = new PdfReader(new ByteArrayInputStream(expectedPdfContent));
 		PdfReader actualReader = new PdfReader(new ByteArrayInputStream(actualPdfContent));
 
 		String expectedText = PdfTextExtractor.getTextFromPage(expectedReader, 1);
 		String actualText = PdfTextExtractor.getTextFromPage(actualReader, 1);
 
-		assert expectedText.equals(actualText) : "PDF content does not match";
+		if (!expectedText.equals(actualText)) {
+			String diffMessage = generateTextDiff(expectedText, actualText);
+			throw new AssertionFailedError(
+					"PDF content does not match for file: " + fileName + "\n" + diffMessage,
+					expectedText, actualText);
+		}
+	}
+
+	/**
+	 * Generate a readable diff between two strings.
+	 * 
+	 * @param expected The expected string
+	 * @param actual   The actual string
+	 * @return A formatted diff message
+	 */
+	private String generateTextDiff(String expected, String actual) {
```

Comment:
There's no need to write your own text differ (although it might be tempting with AI 😁), you can just use https://github.com/java-diff-utils/java-diff-utils and the output will be much more readable.